MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -1176,6 +1176,18 @@ public Date getDate() {
         return personnel;
     }
 
+    /**
+     * Provides a filtered list of personnel including only active Persons.
+     * @return ArrayList<Person>
+     */
+    public ArrayList<Person> getActivePersonnel() {
+        ArrayList<Person> activePersonnel = new ArrayList<Person>();
```

Comment:
Well, at the very least, this "getActivePersonnel()" function is now in a single location, so we can optimize it later on.

Reply:
I imagine the most straightforward way to do that would be to ensure `PersonAddedEvent`, `PersonRemovedEvent` and `PersonChangedEvent` (or perhaps a subclass thereof only for status changes) get fired diligently, and subscribe an updater method to maintain a private activePersonnel list?

Reply:
Yeah, I think that's probably the best approach.

Reply:
I'll add that on my todo list for its own later PR, if no one else gets to it before then.